File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/GuildInvitationStateRecrutedMessage.scala





















// Generated on 05/03/2013 10:34:50
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildInvitationStateRecrutedMessage extends MessageDeserializer {
  val messageId = 5548

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GuildInvitationStateRecrutedMessage.messageId

    val invitationState: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(invitationState)
    }

  }

  protected def create(invitationState: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val invitationState = buf.readByte()

    create(invitationState)
  }

}

object GuildInvitationStateRecrutedMessage extends GuildInvitationStateRecrutedMessage {
  case class MessageImpl(invitationState: Byte) extends Message
  type Target = MessageImpl

  override def create(invitationState: Byte) = MessageImpl(invitationState)

  def apply(invitationState: Byte) = create(invitationState)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildInvitationStateRecrutedMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildInvitationStateRecrutedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5548;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte invitationState;
    
    public GuildInvitationStateRecrutedMessage() { }
    
    public GuildInvitationStateRecrutedMessage(byte invitationState) {
        this.invitationState = invitationState;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(invitationState);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        invitationState = buf.readByte();
        if (invitationState < 0)
            throw new RuntimeException("Forbidden value on invitationState = " + invitationState + ", it doesn't respect the following condition : invitationState < 0");
    }
    
}
